### PR TITLE
Add RML validation rules

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Closes https://github.com/Clinical-Genomics/improve-order-flow/issues/81

### Added

- Validation rules and service for RML

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/order_validation_service/workflows/rml/validation_rules.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    validate_applications_not_archived,
+    validate_container_name_required,
+    validate_sample_names_available,
+    validate_sample_names_unique,
+    validate_tube_container_name_unique,
+    validate_volume_interval,
+    validate_volume_required,
+    validate_well_position_format,
+    validate_well_positions_required,
+    validate_wells_contain_at_most_one_sample,
```
Comment (islean) [inline cg/services/order_validation_service/workflows/rml/validation_rules.py:28]: These will need reworking according to what we decide in https://github.com/Clinical-Genomics/improve-order-flow/issues/98. Least invasive might be some alias handling

File: cg/services/order_validation_service/workflows/rml/validation_rules.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    validate_application_compatibility,
+    validate_application_exists,
+    validate_applications_not_archived,
+    validate_container_name_required,
+    validate_sample_names_available,
+    validate_sample_names_unique,
+    validate_volume_interval,
+    validate_volume_required,
+    validate_well_position_rml_format,
+    validate_well_positions_required,
```
Comment (diitaz93) [inline cg/services/order_validation_service/workflows/rml/validation_rules.py:24]: Regarding this one, should we refactor the https://github.com/Clinical-Genomics/cg/blob/6e180f846cd136d6bd7ecc55b505c425e55bdfe5/cg/services/order_validation_service/rules/sample/utils.py#L26 to comply with the RML plate configuration? I would do that in another PR

File: cg/services/order_validation_service/errors/sample_errors.py [hunk 3]
```diff
@@ -66,6 +71,11 @@ class WellFormatError(SampleError):
     message: str = "Well position must follow the format A-H:1-12"
 
 
+class WellFormatRmlError(SampleError):
+    field: str = "well_position_rml"
+    message: str = "Well position must follow the format A1-H12"
```
Comment (islean) [inline cg/services/order_validation_service/errors/sample_errors.py:76]:
```suggestion
    message: str = "Well position must follow the format A:1-H:12"
```

File: cg/services/order_validation_service/rules/sample/rules.py [hunk 4]
```diff
@@ -208,9 +212,31 @@ def validate_well_position_format(order: OrderWithSamples, **kwargs) -> list[Wel
     return errors
 
 
+def validate_well_position_rml_format(
+    order: OrderWithSamples, **kwargs
```
Comment (islean) [inline cg/services/order_validation_service/rules/sample/rules.py:216]:
```suggestion
    order: OrderWithIndexedSamples, **kwargs
```

File: tests/services/order_validation_service/sample_rules/conftest.py [hunk 5]
```diff
@@ -114,3 +117,35 @@ def valid_fastq_order() -> FastqOrder:
         samples=samples,
         name="FastqOrder",
     )
+
+
+def create_rml_sample(id: int) -> RmlSample:
+    return RmlSample(
+        application="WGSPCFC030",
```
Comment (islean) [inline tests/services/order_validation_service/sample_rules/conftest.py:124]: Fairly certain that this app tag is not valid for RML but we can fix that in a future PR where we add an integration test

**Review by islean — APPROVED** [review]
Looks super!
